draft-nro-bulk-rdap.md

```diff
@@ -48,184 +48,294 @@ feature of this service.
+[NRO RDAP Profile](https://bitbucket.org/nroecg/nro-rdap-profile/raw/v1/nro-rdap-profile.txt) ("nro_rdap_profile_0" as
+of this writing). In the data objects, the Bulk RDAP extension requirements MUST supersede the NRO RDAP Profile
+extension requirements, primarily to afford bulk data compactness.
 
 # Bulk RDAP URL {#bulk_rdap_url}
 
 * Scheme: HTTPS
 * Method: GET
 * Path Segment: nroBulkRdap1?objectClass=<RDAP object class name>
-* Returns: Bulk data as either JSON data ((#bulk_data_format)) or JSON Web Signature (JWS) string
-  ((#security_considerations))
-* Content-Type: application/rdap+json
+* Returns: See (#content_types)
+
+The HTTPS scheme MUST be used for the Bulk RDAP URL.
 
-Limiting the query to a single RDAP object class in the Bulk RDAP URL should result in a simpler service implementation,
-as well as better performance when serving requests.
+When the "objectClass" query parameter is present in the request URL, the bulk data response MUST be limited to all the
+objects for the RDAP object class listed in that query parameter.
 
-Here is an example URL to get bulk data for the IP Network object class:
+When the "objectClass" query parameter is absent in the request URL, the bulk data response MUST return all the objects
+for all the RDAP object classes the server supports.
+
+The following is an example URL to get bulk data for the IP Network object class:
 
 ```
 https://example.net/nroBulkRdap1?objectClass=ip%20network
 ```
 
-_Figure 2_
-
-For a 200 OK response ([@!RFC9110, section 15.3.1]), the server MUST return a JSON object with its "objects" member
-filled with all the objects for the RDAP object class listed in the "objectClass" query parameter.
-
-If the RDAP object class listed in the "objectClass" query parameter is valid ([@!RFC9083, section 5], or a future RDAP
+If the RDAP object class listed in the "objectClass" query parameter is valid ([@!RFC9083, section 5] or a future RDAP
 object class) but the bulk data functionality has not been implemented for it, the server SHOULD return a 501 Not
 Implemented response ([@!RFC9110, section 15.6.2]).
 
 If the RDAP object class listed in the "objectClass" query parameter is invalid, the server SHOULD return a 400 Bad
 Request response ([@!RFC9110, section 15.5.1]).
 
+The following is an example URL to get bulk data for all the RDAP object classes an RIR supports:
+
+```
+https://example.net/nroBulkRdap1
+```
+
+## Content Types {#content_types}
+
+The content type for a bulk data response can be:
+
+* "application/json" -- for a newline-delimited sequence of metadata JSON object and RDAP objects, or for JWS JSON
+  serialization ([@!RFC7515, section 3.2]) of the metadata and data objects (see (#security_considerations))
+* "application/jose" -- for JWS compact serialization ([@!RFC7515, section 3.1]) of the metadata and data objects (see
+  (#security_considerations))
+* "application/gzip" -- for gzip ([@RFC1952]) of either the metadata and data objects or their JWS
```

Comment:
It should be possible to use `Content-Encoding` for compression.

Reply:
Right. Then, the Content-Type for the gzip case should be `application/octet-stream`, no?

Reply:
No, the content type stays unchanged in that case.

Reply:
I think it goes this way if client wants to download a .gz file than we should set  Content-Type as application/gzip 

If client want to download a text file, and the server decides to gzip the content to reduce the size of the data transmitted, than we should set  Content-Encoding: gzip. in this case client will  end up with a text file, uncompressed, as the file's content was only gzipped for transmission.

Reply:
So, looks like we set both content-type `application/gzip` (actual content format part) and content-encoding `gzip` (transmission part) for our purposes, right?

Reply:
No, with those headers the response would have to be compressed twice.  What I think should happen here is that `application/gzip` is removed as a content type, and guidance about using compression is added.  That could be something like "these files are likely to be large, so servers and clients are RECOMMENDED to rely on Content-Encoding in order to reduce the amount of data that is transmitted".

Reply:
Yep, makes sense.

Reply:
> I think it goes this way if client wants to download a .gz file than we should set Content-Type as application/gzip

To Mahesh's point, should we allow `application/gzip` in the Accept header? If so, in this case, we would set content-type to `application/gzip` but not set content-encoding to avoid double compression, right?

Reply:
> If it wants transmission efficiency, then it should also set content-encoding to gzip, right?

That's right.  (Depending on the headers set by the client, the server may not be able to do this, though, in which case it could opt to fall back to an uncompressed response, or (I think) return a 406, effectively saying 'you must support gzip to make this request'.)

> To Mahesh's earlier point, should we allow application/gzip in the Accept header?

I think using content encoding gets to the same outcome, so I would remove it from the document.

Reply:
Ack